tests/README.md

```diff
@@ -20,3 +20,29 @@ https://github.com/vmg/clar
 * Make sure everything is fine.
 
 * Send your pull request. That's it.
+
+
+Memory leak checks
+------------------
+
+These are automatically run as part of CI, but if you want to check locally:
+
+#### Linux
+
+Uses [`valgrind`](http://www.valgrind.org/):
+
+```console
+$ cmake -DBUILD_CLAR=ON -DVALGRIND=ON ..
+$ cmake --build .
+$ valgrind --leak-check=full --show-reachable=yes --num-callers=50 --suppressions=../libgit2_clar.supp \
+  ./libgit2_clar
+```
+
+#### OSX
+
+Uses [`leaks`](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/ManagingMemory/Articles/FindingLeaks.html), which requires XCode installed:
+
+```console
+$ MallocStackLogging=1 MallocScribble=1 MallocLogFile=/dev/null CLAR_AT_EXIT="leaks -quiet \$PPID" \
+  ./libgit2_clar
+```
```

Comment:
> Alternatively, you can setup those environment variables in Xcode's Run phase settings (ie. use the GUI). The only unexposed settings are `MallocLogFile`, `CLAR_AT_EXIT`.

And I've recently noticed that `malloc(3)` has gained (when?) a `MallocDebugReport=none`, which completely silences output and thus seems to be better behaved than `MallocLogFile=/dev/null`. YMMV.

Reply:
> (ie. use the GUI).

eww, someone actually uses XCode? :trollface:  😉

Reply:
Hmm, 10.13.6 doesn't have it, maybe a Mojave feature? I can't find any reference on the Internet either.

Would be good to reduce the noise though — to me a single test test looks like:
```
$ MallocStackLogging=1 MallocScribble=1 MallocDebugReport=none CLAR_AT_EXIT="leaks -quiet \$PPID" ./libgit2_clar -v -smerge::trees::analysis -smerge::workdir::analysis
libgit2_clar(53033,0x7fff898ff380) malloc: stack logs being written into /tmp/stack-logs.53033.10ecce000.libgit2_clar.MiNeOc.index
libgit2_clar(53033,0x7fff898ff380) malloc: recording malloc and VM allocation stacks to disk using standard recorder
libgit2_clar(53033,0x7fff898ff380) malloc: enabling scribbling to detect mods to free blocks
libgit2_clar(53033,0x7fff898ff380) malloc: process 52982 no longer exists, stack logs deleted from /tmp/stack-logs.52982.100cca000.cp.ezBoTL.index
libgit2_clar(53033,0x7fff898ff380) malloc: process 52949 no longer exists, stack logs deleted from /tmp/stack-logs.52949.10191a000.libgit2_clar.fu22PU.index
Loaded 368 suites:
Started (test status codes: OK='.' FAILURE='F' SKIPPED='S')

cp(53034,0x7fff898ff380) malloc: stack logs being written into /tmp/stack-logs.53034.10b41b000.cp.MH8y8U.index
cp(53034,0x7fff898ff380) malloc: recording malloc and VM allocation stacks to disk using standard recorder
cp(53034,0x7fff898ff380) malloc: enabling scribbling to detect mods to free blocks
cp(53034,0x7fff898ff380) malloc: stack logs deleted from /tmp/stack-logs.53034.10b41b000.cp.MH8y8U.index
rm(53035,0x7fff898ff380) malloc: stack logs being written into /tmp/stack-logs.53035.10b9da000.rm.huAWgg.index
rm(53035,0x7fff898ff380) malloc: recording malloc and VM allocation stacks to disk using standard recorder
rm(53035,0x7fff898ff380) malloc: enabling scribbling to detect mods to free blocks
rm(53035,0x7fff898ff380) malloc: stack logs deleted from /tmp/stack-logs.53035.10b9da000.rm.huAWgg.index
merge::trees::analysis.cp(53036,0x7fff898ff380) malloc: stack logs being written into /tmp/stack-logs.53036.106897000.cp.gi6aER.index
cp(53036,0x7fff898ff380) malloc: recording malloc and VM allocation stacks to disk using standard recorder
cp(53036,0x7fff898ff380) malloc: enabling scribbling to detect mods to free blocks
cp(53036,0x7fff898ff380) malloc: stack logs deleted from /tmp/stack-logs.53036.106897000.cp.gi6aER.index
rm(53037,0x7fff898ff380) malloc: stack logs being written into /tmp/stack-logs.53037.104d8a000.rm.nijd8G.index
rm(53037,0x7fff898ff380) malloc: recording malloc and VM allocation stacks to disk using standard recorder
rm(53037,0x7fff898ff380) malloc: enabling scribbling to detect mods to free blocks
rm(53037,0x7fff898ff380) malloc: stack logs deleted from /tmp/stack-logs.53037.104d8a000.rm.nijd8G.index
.```

Reply:
I don't think you can get lower noise than that, apart from that Mojave thing (which I clearly am glad for).

Reply:
Let's keep the documentation as-is for now, since it's nice to keep it in line with the CI steps.  But once Mojave is a little more prevalent, this is going to be great.